src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CryptoStream.cs

```diff
@@ -734,7 +734,6 @@ private async Task CopyToAsyncInternal(Stream destination, int bufferSize, Cance
             finally
             {
                 CryptographicOperations.ZeroMemory(rentedBuffer.AsSpan(0, bufferSize));
```

Comment:
This is now slightly extending the lifetime of the handle. These crypto APIs have been written with specific intent. @bartonjs or @vcsjones Are you okay with this change in lifetime of the GCHandle?

Reply:
I don't think the lifetime extension is a big concern. The point of the pin is to make sure the GC doesn't copy it around (move it). As long as the `ZeroMemory` happens before the pin is let go, I think it's fine. However:

1. It seems a little odd to return a rented buffer while it is still pinned.
2. `System.Security` prefers using explicit scopes for `using`.

Reply:
Or, more specifically, since there is a brief window of time while a pinned buffer has been returned to the pool, what happens if thread A returns a pinned buffer, thread B rents it and tries to pin it, then thread A unpins it?

It seems like the most sensible thing to do is just make sure it unpinned before it gets returned.

Reply:
Shouldn't multiple pinning handles working simultaneously? In other words, when there's at least 1 pinning handle, the array should be pinned.
Agree that the pinning period should just over clearing.

Reply:
Yeah, this should just undo the `using` and keep the Free/Dispose where it was.  It's already a try/finally, we don't need a try/finally wrapping the try/finally.

Reply:
`using` multiple variables would also create nested try/finally. It should be acceptable pattern.

Reply:
@huoyaoyuan Please revert this change by the request of the code owner, @bartonjs.

Reply:
The `using` specifically. I think the nullable change is appropriate.